lib/wasi-types/src/wasi/bindings.rs

```diff
@@ -2328,6 +2328,93 @@ impl core::fmt::Debug for Signal {
         }
     }
 }
+
+#[repr(u8)]
+#[derive(Clone, Copy, PartialEq, Eq, num_enum :: TryFromPrimitive, Hash)]
+pub enum SigAction {
+    Default,
+    Ignore,
+}
+impl core::fmt::Debug for SigAction {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            SigAction::Default => f.debug_tuple("SigAction::Default").finish(),
+            SigAction::Ignore => f.debug_tuple("SigAction::Ignore").finish(),
+        }
+    }
+}
+
+#[doc = " A signal and its corresponding action."]
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct SignalAndAction {
```

Comment:
I think, based on the rest of the code, it may be more consistent to name `SignalAndAction` -> `Signal`, and then the current `Signal` name it `SignalKind`